infra_event_notifier/notifier.py

```diff
@@ -1,47 +1,28 @@
-from typing import Mapping
-
-from infra_event_notifier.backends.datadog import send_event
-from infra_event_notifier.backends.jira import create_issue
-from infra_event_notifier.backends.slack import send_notification
+from typing import Self
 
 
 class Notifier:
-    def __init__(
-        self,
-        datadog_api_key: str | None = None,
-        slack_api_key: str | None = None,
-        jira_api_key: str | None = None,
-    ) -> None:
-        self.datadog_api_key = datadog_api_key
-        self.slack_api_key = slack_api_key
-        self.jira_api_key = jira_api_key
+    def __init__(self) -> None:
+        # Notification fields
+        self.title: None | str = None
+        self.body: None | str = None
```

Comment:
As title and body are always supposed to be present, please add them to the init rather than having setter. 
In a builder pattern, setters are useful only when we have an object that can be initialized in different ways. If all setters have to be called each time, the builder is not useful.